File: OpenMP_cpu/simd/1_ptrs/ptrs.f90
subroutine add_floats(a,b,c,d,e,n)
real,pointer,dimension(:) :: a,b,c,d,e
!   Note, a,b,c,d and e are pointers
!   With 5 pointers, compiler gives up on vectorization
!   Need to force the compiler to SIMDize it.

!   !$omp ??
    do i = 1,n
       a(i) = a(i) + b(i) + c(i) + d(i) + e(i)
    end do
end subroutine

File: OpenMP_cpu/simd/1_ptrs/solutions/ptrs.c
void add_floats(float *a, float *b, float *c, float *d, float *e, int n) {
//   With 5 pointers, compiler gives up on vectorization
//   Need to force the compiler to SIMDize it.
   int i;
    #pragma omp simd
    for (i=0; i<n; i++)
       a[i] = a[i] + b[i] + c[i] + d[i] + e[i];
}
